### PR TITLE
Add Report Details UI

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
#4 
*Description of changes:*
Adding finalized UI and snapshot testing for Report Details page. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/components/main/report_details.tsx [hunk 1]
```diff
@@ -34,20 +29,25 @@ import {
...
+    EuiDescriptionListDescription,
   } from '@elastic/eui';
 
   interface RouterHomeProps {
     httpClient?: any,
   }
 
-  export class ReportDetails extends React.Component<RouterHomeProps> {
+  interface ReportDetailsState {
+    eventId: number
```
Comment (seraphjiang) [inline public/components/main/report_details.tsx:42]: reportId?

File: public/components/main/report_details.tsx [hunk 2]
```diff
@@ -34,20 +29,25 @@ import {
...
+  }
+
+  export class ReportDetails extends React.Component<RouterHomeProps, ReportDetailsState> {
     constructor(props) {
       super(props);
-      this.state = {};
+      this.state = {
+        eventId: this.props.match.params.eventId,
```
Comment (seraphjiang) [inline public/components/main/report_details.tsx:49]: reportId?

File: public/components/main/router_home.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ export class RouterHome extends React.Component<RouterHomeProps, any> {
                 <EuiPageContentBody>
                   <Switch>
                     <Route
-                      path="/report_details/:id"
+                      path="/report_details/:eventId"
```
Comment (seraphjiang) [inline public/components/main/router_home.tsx:53]: reportId?

**Comment by davidcui-amzn** [comment]
Screen Shot 2020-06-30 at 2 19 36 PM
Screen Shot 2020-06-30 at 2 20 52 PM

First screenshot is current old mock-up, second one is new finalized one for `Report Details`. Since finalized Details page has been completed, do you recommend updating this PR to reflect that one @seraphjiang

**Comment by seraphjiang** [comment]
> Screen Shot 2020-06-30 at 2 19 36 PM
> Screen Shot 2020-06-30 at 2 20 52 PM
> 
> First screenshot is current old mock-up, second one is new finalized one for `Report Details`. Since finalized Details page has been completed, do you recommend updating this PR to reflect that one @seraphjiang

Always good to keep document update to date, but should be ok to wait for code complete.

File: public/components/main/report_details/report_details.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                      <h1>{reportDetailsMetadata.report_name}</h1>
+                    </EuiTitle>
+                  </EuiPageHeaderSection>
+                </EuiFlexItem>
+              <EuiFlexGroup justifyContent="flexEnd" alignItems="flexEnd" gutterSize="xs">
+                <EuiFlexItem></EuiFlexItem>
+                <EuiFlexItem></EuiFlexItem>
+                <EuiFlexItem></EuiFlexItem>
+                <EuiFlexItem></EuiFlexItem>
+                <EuiFlexItem></EuiFlexItem>
```
Comment (seraphjiang) [inline public/components/main/report_details/report_details.tsx:135]: why so many empty EuiFlexItem, would you try https://elastic.github.io/eui/#/layout/flex. would you try ```Flex item can specify a proportional width```

Comment (davidcui-amzn) [reply public/components/main/report_details/report_details.tsx:135]: Empty EuiFlexItems were to make sure the `Archive` `Share` and `Download` texts were spaced properly, had trouble with just specifying with the FlexItem

**Pull request merged**